Feature: specify compiler cpp std with cpp language standard

tarc opened the pull request:
Take the value set for C++ Language Standard, in the project's properties page, and use it to define a `-s compiler.cppstd` setting during conan install.

tarc committed:
Map LanguageStandard into CppStd

tarc committed:
Pass cppstd setting to conan install

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/conan-io/conan-vs-extension?pullRequest=180) <br/>All committers have signed the CLA.

tarc committed:
Try get LanguageStandard by many ways

Try at first get it with VCCLCompilerTool and then with CL and
ConfigurationGeneral rules. If none of them works, do not pass the
cppstd setting argument to conan install.

tarc commented:
fixes #181

tarc commented:
This patch would really help me selling conan to my team. Is there something that I can help you guys with in order to make it upstream?

@jgsogo @SSE4

tarc committed:
Fix bug when LanguageStandard is set as "Default"

AbrilRBS commented:
Closing as we've just released a new vs2022 compatible version that differs from this base, which will be the version we'll be supporting in the future, feel free to check it out in https://marketplace.visualstudio.com/items?itemName=conan-io.conan-vs-extension&ssr=false#overview
